migrations/V42__Plan_Favorite_To_User_Table.sql

```diff
@@ -0,0 +1,58 @@
+/* ADD Temp data column for this */
+
+ALTER TABLE plan_favorite
+RENAME COLUMN user_id TO user_id_old;
+
+ALTER TABLE plan_favorite
+RENAME COLUMN created_by TO created_by_old;
+
+
+ALTER TABLE plan_favorite
+RENAME COLUMN modified_by TO modified_by_old;
+
+/* ADD Correct Column */
+ALTER TABLE plan_favorite
+ADD COLUMN user_id UUID REFERENCES public.user_account (id) MATCH SIMPLE,
+ADD COLUMN created_by UUID REFERENCES public.user_account (id) MATCH SIMPLE,
+ADD COLUMN modified_by UUID REFERENCES public.user_account (id) MATCH SIMPLE;
+
+/* Perform the data migration */
+WITH userAccount AS (
+    SELECT
+        plan_favorite.id AS favoriteID,
+        plan_favorite.model_plan_id,
+        plan_favorite.user_id_old,
+        user_account.id AS user_id,
+        user_account_created.id AS created_by,
+        user_account_modified.id AS modified_by
+    FROM plan_favorite
+    LEFT JOIN user_account ON plan_favorite.user_id_old = user_account.username
+    LEFT JOIN user_account AS user_account_created ON plan_favorite.created_by_old = user_account_created.username
+    LEFT JOIN user_account AS user_account_modified ON plan_favorite.modified_by_old = user_account_modified.username
+)
+
+UPDATE plan_favorite
+SET
+    created_by = userAccount.created_by,
+    modified_by = userAccount.modified_by,
+    user_id = userAccount.user_id
+FROM userAccount
+WHERE userAccount.favoriteID = plan_favorite.id;
```

Comment:
Thanks @ClayBenson94! Yes, because of the recent ticket to always create a user account whenever is added as a collaborator, there _should_ be an account for every plan favorite. Because of that, I'd actually want this to fail if it wasn't the case so the data isn't stuck in a partial migration. Thoughts?

Reply:
That makes sense!

I presume we could also do a data wipe in each env to ensure that everything's as expected, as I know we've deployed quite a few migrations to each env

Reply:
Thanks Clay! That is a good call, and chances that there might be some collaborators without an account due to the timing of the original data backfill. I think it would probably be a good test to run this as is to test the approach first and see if there are any issues. 

If there do seem to be issues, I think we could wipe the DB and try again. Thoughts?

Reply:
@ClayBenson94 , I had one more thought about how to handle the actual migration to the user table. 

We could write something so it will create a dummy user account record if there happens to be any EUAIDs that don't match a user account. This would not have any real information in it except for the EUAID. We'd set the hasLoggedIn property to false, which would trigger the data getting updated on the first login. Thoughts?

Reply:
That actually seems like it could be a good fallback - if it's not too difficult to write, let's do it! Otherwise, sounds like we have confidence in the data.